docs/getting_started/parametrising_components.md

```diff
@@ -68,46 +74,40 @@ from django_components import Component, register
 class Calendar(Component):
     template_file = "calendar.html"
     ...
-    def get_context_data(self, date: date, extra_class: str = "text-blue"):
+    def get_template_data(self, args, kwargs, slots, context):
         return {
-            "date": date,
-            "extra_class": extra_class,
+            "date": kwargs["date"],
+            "extra_class": kwargs.get("extra_class", "text-blue"),
         }
 ```
 
-!!! info
-
-    Since [`get_context_data()`](../../reference/api#django_components.Component.get_context_data)
-    is just a regular Python function, type hints annotations work the same way as anywhere else.
-
-!!! warning
-
-    Since [`get_context_data()`](../../reference/api#django_components.Component.get_context_data)
-    is just a regular Python function, it will raise TypeError if it receives incorrect parameters.
-
 Since `extra_class` is optional in the function signature, it's optional also in the template.
 So both following calls are valid:
 
 ```htmldjango
-{% component "calendar" "2024-12-13" / %}
-{% component "calendar" "2024-12-13" extra_class="text-red" / %}
+{% component "calendar" date="2024-12-13" / %}
```

Comment:
Slight difference is that with `get_context_data()`, one could pass a value (e.g. a `date`) EITHER as arg or kwarg.

But with `get_template_data()`, the args and kwargs are strictly separated.

...Actually, I've just updated the docs to mention this in more detail:

<img width="1199" alt="Screenshot 2025-05-03 at 11 41 47" src="https://github.com/user-attachments/assets/4325c588-d6df-4bee-aefd-576acf415225" />